### DISCUSSION TITLE
Quality Assurance Testing

### DISCUSSION THREAD

**Discussion by mrSeoD** [electricitymaps/electricitymaps-contrib#5267] [Q&A] [answered]
I am getting started on contributing to the frontend side of this project. 
And I was wondering if I could be informed of quality assurance practices (documents, unit test program files, etc.) for this project?
I read through many documents, but I wasn't able to find them. 

Thank you for your help in advance!

**Comment by VIKTORVAV99** [comment, marked as answer]
We use a combination of a few tools for testing.
We use vitest for unit testing and cypress for both component testing and E2E testing.
Then we also have a combination of prettier, eslint, jsonlint and more to make sure the files them selves are formatted and use the correct syntax.

All of these are also automated in our CI and should be pretty straight forward to use (they all have scripts in the package.json).

**Reply by mrSeoD** [reply]
Got it! Thank you for the explanation!
